buildpack/build.go

```diff
@@ -147,17 +192,17 @@ func (b *Descriptor) runBuildCmd(bpLayersDir, bpPlanPath string, config BuildCon
 }
 
 func (b *Descriptor) setupEnv(buildEnv BuildEnv, layersDir string) error {
```

Comment:
Right now, we decode each toml file at least 2 times - once as part of `checkTypesFormat`, and at least once in this function (`setupEnv`).
I'm wondering if we can merge this function with `checkTypesFormat` so we decode each toml file only once. Then we won't need that `isBuild` will decode the file, we'll just check the `Build` flag in the struct.
@ekcasey, any thoughts?

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/9bd6a05415c17939ad6fa3105eae95823aac568f.